### ISSUE TITLE
Nix can't find pkgs.nixos-boot - and other current issues

### ISSUE THREAD

**Issue by codeiimon** [Melkor333/nixos-boot#11]
While following your guide (both the flake and non-flake option) I get this error:

```
/nix/store/...-source/modules.nix:38:23:

           37|     enable = true;
           38|     themePackages = [ pkgs.nixos-boot ];
             |                       ^
           39|     theme = config.nixos-boot.theme;

```

Not sure what is going on 😕

**Comment by Melkor333** [comment]
Hm that should be covered with the overlay in the `modules.nix` 🤔 
But I guess the module is a bit unnecessary complicated anyway...

I pushed a change that makes it not use `overlays` anymore and therefore not use `pkgs.nixos-boot`.
It seems to work, though I think I have some unrelated firmware issue that makes it freeze for me.

Also, I refactored it & added a `evil-nixos` theme which I can't test due to my issues 😅
(I'm having this issue IIRC since fall last year and no clue how to debug, the system just locks up)

It does work in the test-vm, though, so 🤞🏼 my issues are local.
Would be glad if you could test it (and maybe test the `evil-nixos` theme as well if it works 😇 )

**Comment by codeiimon** [comment]
Sorry for the late reply! Didn't get around to testing it till now.

So it works, kinda 😅 ? I don't get the previous error, but when I boot up the machine, instead of logo, I just get a blank white background. Not sure what is happening there.

Also thanks for the quick response. Ill try to respond as fast as possible from now on.

**Comment by Melkor333** [comment]
No worries, anything sub 2 weeks is super fast in OSS :)

Hm okay, I get a similar experience. But with the test vm it just works... I have no clue how to fix this. 😬 
I might've to take a deeper look at the generated script, maybe there is a typo or something...

**Comment by Melkor333** [comment]
@codeiimon for how long do you get the white screen? just a short flash or for a few seconds?

I just set `boot.initrd.systemd.enable = true;` and now it shows me the `evil-nixos` theme while waiting for my LUKS passphrase. So the TODOS are roughly:
- [ ] Figure out why the delay doesn't work (it ignores my 10 second delay I configured)
- [ ] Colors don't work either (at least the `evil-nixos` theme seems to always have white background.)
- [ ] upscale the evil-nixos images, it's quite small right now. Maybe AI can help me with this 🤔 
- [ ] Fix Text color. I see no text when I have to enter my LUKS password 😄 
- [ ] document possible use of `boot.initrd.systemd.enable` to not have the ugly LUKS prompt (only learned about this right now)

**Comment by codeiimon** [comment]
Yeah I just got around to testing it again (sorry for the late reply again, last weeks of school can be alot), and my experience is the same as yours. I can't really test the `boot.initrd.systemd.enable =true` though as I'm not using LUKS and don't plan too. Plus I also don't want to have to enable it as I'm using grub2 which is of course incompatible.

Anyway thank you so much for the work! I'm excited to see this work.
